Content.Server/GameObjects/Components/Weapon/Ranged/Projectile/AmmoWeapon.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Content.Server.GameObjects.Components.Sound;
+using Robust.Server.GameObjects.Components.Container;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System;
+
+namespace Content.Server.GameObjects.Components.Weapon.Ranged.Projectile
+{
+    /// <summary>
+    ///     Handles firing projectiles from a contained <see cref="BallisticBulletComponent" />.
+    /// </summary>
+    public abstract class AmmoWeaponComponent : SharedProjectileWeaponComponent
+    {
+        protected Chamber[] Chambers;
```

Comment:
Put this behind a `private set;` property please. So that somebody can't re-assign it to have a different count from `ChamberCount`.